### PR TITLE
E2E test for nginx ssl

### PR DESCRIPTION
**Pull request by Damenus**

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -410,6 +411,58 @@ pipeline {
...
+                    KUBECONFIG="${HOME}/.kube/admin.conf"
+                    KUSTOMIZATION_MONITORING='examples/kubernetes/monitoring/'
+                    KUSTOMIZATION_WORKLOAD='examples/kubernetes/workloads/'
+                    WCA_SCHEDULER_PATH='examples/kubernetes/wca-scheduler/'
+                }
+                steps {
+
+                    print('Set configs wca-wcheduler...')
+                    sh "sed -i 's#/var/run/secrets/kubernetes.io/serviceaccount/ca.crt;#/var/run/secrets/kubernetes.io/cert/CA.crt;#g' ${WORKSPACE}/${WCA_SCHEDULER_PATH}wca-scheduler-server.conf"
+                    sh "sed -i 's/node36/node18/g' ${WORKSPACE}/${WCA_SCHEDULER_PATH}wca-scheduler-deployment.yaml"
```
Comment (wacuuu) [inline Jenkinsfile:430]: Maybe parameterize last IP part?

File: Jenkinsfile [hunk 2]
```diff
@@ -410,6 +411,58 @@ pipeline {
...
+                        --from-file ${WORKSPACE}/tests/e2e/nginx/server.crt \
+                        --from-file ${WORKSPACE}/tests/e2e/nginx/server-key.pem \
+                        --from-file ${WORKSPACE}/tests/e2e/nginx/CA.crt"
+
+                    print('Starting prometheus...')
+                    sh "kubectl apply -k ${WORKSPACE}/${KUSTOMIZATION_MONITORING}prometheus/"
+
+                    print('Starting wca-wcheduler...')
+                    sh "kubectl apply -k ${WORKSPACE}/${WCA_SCHEDULER_PATH}"
+                    sleep 15
```
Comment (wacuuu) [inline Jenkinsfile:443]: Status check after this sleep to see if the cluster is ready?

File: Jenkinsfile [hunk 3]
```diff
@@ -410,6 +411,58 @@ pipeline {
...
+                        deactivate"
+
+                }
+                post {
+                    always {
+                        print('Cleaning wca-scheduler...')
+                        sh "kubectl delete -k ${WORKSPACE}/${WCA_SCHEDULER_PATH} --wait=false"
+                        sh "kubectl delete secret  wca-scheduler-cert -n wca-scheduler"
+                        sh "kubectl delete svc wca-scheduler-nodeport-service --namespace wca-scheduler"
+                        junit 'unit_results.xml'
```
Comment (wacuuu) [inline Jenkinsfile:462]: This is not a cleanup; deserves a separate print('Asserting unit tests status...') or something.

File: tests/e2e/nginx/test_wca_nginx_ssl.py [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+    cert = (_CERT_FILE_PATH.format(workspace=workspace, case=cert_name),
+            _KEY_FILE_PATH.format(workspace=workspace, case=cert_name))
+    r = requests.get(_URL.format(ip=ip, port=port), cert=cert, verify=False)
+    assert "true" in r.text
+
+
+@pytest.mark.parametrize('cert_name', [
+    'attribut-expiration-date-past',
+    'attribut-expiration-date-priori',
```
Comment (wacuuu) [inline tests/e2e/nginx/test_wca_nginx_ssl.py:45]: What is the context for this name?

File: tests/e2e/nginx/test_wca_nginx_ssl.py [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+    'attribut-purpose',
```
Comment (wacuuu) [inline tests/e2e/nginx/test_wca_nginx_ssl.py:46]: typoo; attribut -> attribute

File: Jenkinsfile (hunk 1 above)
Comment (Damenus) [reply Jenkinsfile:430]: Done

File: Jenkinsfile (hunk 3 above)
Comment (Damenus) [reply Jenkinsfile:463]: Done. You're right.

File: tests/e2e/nginx/test_wca_nginx_ssl.py (hunk 5 above)
Comment (Damenus) [reply tests/e2e/nginx/test_wca_nginx_ssl.py:46]: Fixed

File: tests/e2e/nginx/test_wca_nginx_ssl.py (hunk 4 above)
Comment (Damenus) [reply tests/e2e/nginx/test_wca_nginx_ssl.py]: I changed `piori` to `issued-too-early`

File: Jenkinsfile (hunk 2 above)
Comment (Damenus) [reply Jenkinsfile]: I added function `wait_for_wca_wcheduler` to check this

**Pull request merged**